www/caddy/src/opnsense/mvc/app/models/OPNsense/Caddy/Caddy.php

```diff
@@ -98,28 +98,33 @@ private function checkWebGuiSettings($messages)
         // Get custom caddy ports if set. If empty, default to 80 and 443.
         $httpPort = !empty((string)$this->general->HttpPort) ? (string)$this->general->HttpPort : '80';
         $httpsPort = !empty((string)$this->general->HttpsPort) ? (string)$this->general->HttpsPort : '443';
-        $tlsAutoHttpsSetting = (string)$this->general->TlsAutoHttps;
 
-        // Check for conflicts
         $overlap = array_intersect($this->getWebGuiPorts(), [$httpPort, $httpsPort]);
 
-        if (!empty($overlap) && $tlsAutoHttpsSetting !== 'off') {
+        if (
+            !empty($overlap) &&
+            !empty((string)$this->general->TlsEmail) &&
+            empty((string)$this->general->TlsAutoHttps)
+        ) {
             $portOverlap = implode(', ', $overlap);
+
+            $validationMessage = gettext(
+                'Resolve conflicting HTTP and/or HTTPS port(s) with ' .
+                'the OPNsense WebGUI: "%s".'
+            );
+
             $messages->appendMessage(new Message(
-                sprintf(
-                    gettext(
-                        'To use "Auto HTTPS", resolve these conflicting ports %s ' .
-                        'that are currently configured for the OPNsense WebGUI. ' .
-                        'Go to "System - Settings - Administration". ' .
-                        'To release port 80, enable "Disable web GUI redirect rule". ' .
-                        'To release port %s, change "TCP port" to a non-standard port, ' .
-                        'e.g., 8443.'
-                    ),
-                    $portOverlap,
-                    $httpsPort
-                ),
+                sprintf($validationMessage, $portOverlap),
                 "general.TlsAutoHttps"
             ));
+            $messages->appendMessage(new Message(
+                sprintf($validationMessage, $portOverlap),
+                "general.HttpPort"
```

Comment:
adding the same to all fields kind of defeats the purpose of having specific validation messages telling people which value they have to change in order to proceed :)

Reply:
This whole WebGUI validation is a real burden and there is almost no benefit in it anymore by how lenient it became.

It is also quite hard to tell the user what exactly to fix at what spot.

Maybe the cleanest would be to remove it completely, since other services also don't warn you when there is a port conflict.

Reply:
At the same time the docs clearly state what to do:

https://docs.opnsense.org/manual/how-tos/caddy.html#prepare-opnsense-for-caddy-after-installation

And they're referenced in the pkg description if people want to read:

https://github.com/opnsense/plugins/blob/7a1dce77d632d0a1d4f876da858e38cdee3fef72/www/caddy/pkg-descr#L11